src/callbacks/callback_gpu_memory_usage.cpp

```diff
@@ -27,6 +27,30 @@
 
 #include "lbann/callbacks/callback_gpu_memory_usage.hpp"
 #include <iomanip>
+#include <sstream>
+
+namespace {
+template <typename T>
+T get_mean(const std::vector<T> &v) {
+  return std::accumulate(v.begin(), v.end(), 0.0) /
+      v.size();
+}
+template <typename T>
+T get_median(const std::vector<T> &v) {
+  std::vector<T> tmp = v;
+  std::sort(tmp.begin(), tmp.end());
```

Comment:
You might consider using `std::nth_element` to do only a partial sort instead of completely sorting.